src/native/metrics/EventLoop.cpp

```diff
@@ -0,0 +1,45 @@
+#include "EventLoop.hpp"
+
+namespace datadog {
+  // http://docs.libuv.org/en/v1.x/design.html#the-i-o-loop
+  EventLoop::EventLoop() {
+    uv_check_init(uv_default_loop(), &check_handle_);
+    uv_unref(reinterpret_cast<uv_handle_t*>(&check_handle_));
+
+    check_handle_.data = (void*)this;
+    check_usage_ = usage();
+  }
+
+  void EventLoop::check_cb (uv_check_t* handle) {
+    EventLoop* self = (EventLoop*)handle->data;
+    self->tick();
+  }
+
+  void EventLoop::tick () {
+    uint64_t usage = this->usage();
+
+    histogram_.add(usage - check_usage_);
+    check_usage_ = usage;
+  }
+
+  void EventLoop::enable() {
+    uv_check_start(&check_handle_, &EventLoop::check_cb);
+  }
+
+  void EventLoop::disable() {
+    uv_check_stop(&check_handle_);
```

Comment:
I think it would be safer to also add this to destructor. Otherwise `uv_check` might call a callback on freed instance of `EventLoop`.